### PR TITLE
NRPT-13 Add NRCED CSV Export

### PR DESCRIPTION
**Pull request by BcGovNeal**
https://bcmines.atlassian.net/browse/NRPT-13

- Added CSV export to NRCED.  Export button only shows if the results are being filter by keywords or the advanced filters.
- Added `compression` module to the API to enable gzip compression.  API responses for the CSV exports are around 6-8MB per request.  The `compression` module reduces these requests down to 500-700KB.

### CODE REVIEW

File: angular/projects/public-nrpti/src/app/records/utils/record-utils.ts [hunk 1]
```diff
@@ -43,4 +45,188 @@ export class RecordUtils {
...
+  // Escape commas
+  if (str.indexOf(',') > -1) {
+    str = `"${str}"`;
+  }
+
+  return str;
+}
+
+/**
+ * Escape special characters in the CSV fields so the final output can be opened
```
Comment (mtCarto) [inline angular/projects/public-nrpti/src/app/records/utils/record-utils.ts:218]: Update doc description for download

**Review by mtCarto — APPROVED** [review]
Exports look correct, functions as expected.

File: angular/projects/public-nrpti/src/app/records/utils/record-utils.ts (hunk 1 above)
Comment (BcGovNeal) [reply angular/projects/public-nrpti/src/app/records/utils/record-utils.ts]: Fixed in https://github.com/bcgov/NRPTI/pull/798/commits/c3308888b1560b725839ee6c3de47c46c2c87768

**Review by marklise — APPROVED** [review]
👍

**Pull request merged**